tests/unit/capi/GEOSRelatePatternTest.cpp

```diff
@@ -45,5 +45,26 @@ void object::test<11>()
     ensure_equals("curved geometry not supported", GEOSRelatePattern(geom2_, geom1_, "0********"), 2);
 }
 
+// invalid DE-9IM
+template<>
+template<>
+void object::test<3>()
+{
+    geom1_ = fromWKT("POINT(1 2)");
+    geom2_ = fromWKT("POINT(1 2)");
+
+    ensure(geom1_);
+    ensure(geom2_);
+
+    // pattern too long
+    ensure_equals(2, GEOSRelatePattern(geom1_, geom2_, "0FFFFF2120000000000000000000"));
+
+    // pattern too short
+    ensure_equals(2, GEOSRelatePattern(geom1_, geom2_, "0F"));
+
+    // pattern has invalid characters
+    ensure_equals(0, GEOSRelatePattern(geom1_, geom2_, "123456789"));
```

Comment:
@dbaston: is it possible the args to `ensure_equals` are in the wrong order here? (as per https://github.com/libgeos/geos/blob/main/tests/unit/tut/tut_assert.hpp#L159)

Reply:
Yes, as with many of our tests